lib/message-queue.js

```diff
@@ -20,7 +19,7 @@ function MessageQueue() {
 
   self.registerWorker = function(type, promise) {
     _workers[type] = promise;
-    _startPolling();
+    return _startPolling();
```

Comment:
Why the added `return`?

Reply:
`_startPolling` now returns the promise that completes when `createIndex` is done. Someone might care to wait for that?  I don't know; I don't think I have a very good reason for this.  I'm fine with removing it.

Reply:
Ahh, ok. That kind of makes sense.  I guess that is the problem with the async syntax.  It isn't immediately obvious that there is something to return.

Reply:
I'll rewrite it in TypeScript too so that's clearer ;) 

...Alright, I'll stop with the TS nagging

...for tonight